src/main/java/org/jabref/gui/ReplaceStringViewModel.java

```diff
@@ -0,0 +1,117 @@
+package org.jabref.gui;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+
+import org.jabref.gui.undo.NamedCompound;
+import org.jabref.gui.undo.UndoableFieldChange;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+
+public class ReplaceStringViewModel extends AbstractViewModel
+{
+    private boolean allFieldReplace;
+    private String findString;
+    private String replaceString;
+    private String[] fieldStrings;
+    private BasePanel panel;
+
+    private StringProperty findStringProperty = new SimpleStringProperty();
+    private StringProperty replaceStringProperty = new SimpleStringProperty();
+    private StringProperty fieldStringProperty = new SimpleStringProperty();
+    private BooleanProperty allFieldReplaceProperty = new SimpleBooleanProperty();
+    private BooleanProperty selectOnlyProperty = new SimpleBooleanProperty();
+
+
+    public ReplaceStringViewModel(BasePanel basePanel)
+    {
+        this.panel = basePanel;
+    }
+
+    public int replace() {
+        findString = findStringProperty.getValue();
+        replaceString = replaceStringProperty.getValue();
+        fieldStrings = fieldStringProperty.getValue().split(";");
+        boolean selOnly = selectOnlyProperty.getValue();
+        allFieldReplace = allFieldReplaceProperty.getValue();
+
+        final NamedCompound compound = new NamedCompound(Localization.lang("Replace string"));
+        int counter = 0;
+        if (this.panel == null)
```

Comment:
Is this check necessary? You always pass the basePanel as constructor argument.
I would suggest you add an Objects.requireNonNull(basePanel) in the constructor to ensure that the argument is never null. If it is null, it will directly throw and exception which makes it easier to see the real reason,

Reply:
Yes, you are right :), I forgot to check my logic after converting to a viewModel ;-). Thank You!